src/contracts.rs

```diff
@@ -304,40 +443,288 @@ impl TryAsContract for ObjectValidation {
                 required,
                 properties,
                 is_open_record(additional_properties.as_deref()),
+                root_schema,
                 refs_usage,
             ))),
+            // Dicts that have a regex pattern on their fields.
+            (
+                ObjectValidation {
+                    max_properties,
+                    min_properties,
+                    required,
+                    properties,
+                    pattern_properties,
+                    additional_properties: _,
+                    property_names: None,
+                },
+                Some(Schema::Bool(false)),
```

Comment:
What does a `false` here mean?

Reply:
That there are no additional properties (I updated the comment to hopefully be clearer).

The point is that if additional properties are disallowed then we can translate `"propertyNames": { "[a-z]+": schema }` to `{_: schema}` and `FieldsMatch "[a-z]+"`. But if additional properties are allowed then it's more complicated and we fall back to the general case.